### ISSUE TITLE
this package makes `SetMaxIdentifierLength()` doesn't work

### ISSUE THREAD

**Issue by ieyj** [efcore/EFCore.NamingConventions#84]
I use the 1.1.1 version in a asp.net core 3.1 project.

Since I need to deal with a Oracle 11 database, the db table column name case need to be upper to easy to use, I find this package,
But quickly I ran into another problem, the identifier is too long, By Google I find the solution is add `modelBuilder.Model.SetMaxIdentifierLength(30);` to limit the identifier in 30 chars.

However this solution takes no effect in my project. Soon I find after removing the package, the solution take effect.

That's all.

**Comment by ieyj** [comment]
And another small problem.

For example, I have a `public DbSet<Stdudent> Students { get; set; }` in my dbcontext, and I specify a new name to the stduent table name `Student` but not use the default `Students` with `modelBuilder.Entity<Student>().ToTable("Student")`. If I have some key or index related to the `Student` table, ef will name it like `pk_stduent`. But after add this package, this will be named as `pk_students`. Although there is only one letter 's' diffs, I think it shows that there is a problem with the execution order in `EfCore.Naming Conversion`. It was executed prematurely. The right way is naming conversions should always be excuted at last.

**Comment by ieyj** [comment]
Finally I put this code block at the end of `OnModelCreating()` method and it works better than this package for me.

```json
modelBuilder.Model.SetMaxIdentifierLength(30);

foreach (var entityType in modelBuilder,Model.GetEntityTypes())、
{
    foreach (var property in entityType,GetProperties())
    {
        property.SetColumName(property.GetColumnName().ToUpper());
    }

    // same way to set key name, foreignkey name and index name.
}
```

I must say this package is really magic and amazing with just a single line code without these problems.

**Comment by roji** [comment]
Re your 2nd problem, I could not reproduce this with the latest version of the code (about to be released as 6.0.0). With `DbSet<Student> Students`, and an explicit name configuration of the table to `Student`, the primary key is named `pk_student` as expected.

It's possible there was a bug that was already fixed since you filed this - please try the 6.0 version when it's out and open a new issue with the full code sample which shows the problem.

**Comment by roji** [comment]
Re the identifier length, I'm not sure setting SetMaxIdentifierLength in OnModelCreating is supported, and does not seem to work fully. In my code sample below, the PK name indeed gets truncated to 10 chars, but the table name does not. Setting the identifier length is something that's meant for providers, not end users - and setting it in OnModelCreating is likely too late to affect some things which happen before that (this is also probably why EFCore.NamingConventions isn't affected by it).

To summarize, it's the Oracle provider's responsibility to set the max identifier length correctly - it's not something with which you're supposed to concern yourself as a user. If you use the same model with two providers (Oracle and PG), each provider should apply its limits; no need for your PG identifier to be short just because Oracle needs them to be.

<details>
<summary>Code sample</summary>

```c#
await using var ctx = new BlogContext();
await ctx.Database.EnsureDeletedAsync();
await ctx.Database.EnsureCreatedAsync();

public class BlogContext : DbContext
{
    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        => optionsBuilder
            .UseSqlServer(@"Server=localhost;Database=test;User=SA;Password=Abcd5678;Connect Timeout=60;ConnectRetryCount=0")
            .LogTo(Console.WriteLine, LogLevel.Information)
            .EnableSensitiveDataLogging();
            // .UseSnakeCaseNamingConvention()

    public DbSet<Student> Studentsx123456789x123456789x123456789x123456789x { get; set; }

    protected override void OnModelCreating(ModelBuilder modelBuilder)
    {
        modelBuilder.Model.SetMaxIdentifierLength(10);
    }
}

public class Student
{
    public int Id { get; set; }
    public string Name { get; set; }
}
```

</details>
